cql3/statements/create_keyspace_statement.cc

```diff
@@ -148,36 +148,29 @@ std::optional<sstring> check_restricted_replication_strategy(
-    // protective we are configured, let's prevent it or allow with a warning:
-    if (replication_strategy == "org.apache.cassandra.locator.SimpleStrategy") {
-        switch(qp.db().get_config().restrict_replication_simplestrategy()) {
-        case db::tri_mode_restriction_t::mode::TRUE:
-            throw exceptions::configuration_exception(
-                "SimpleStrategy replication class is not recommended, and "
-                "forbidden by the current configuration. Please use "
-                "NetworkToplogyStrategy instead. You may also override this "
-                "restriction with the restrict_replication_simplestrategy=false "
-                "configuration option.");
-        case db::tri_mode_restriction_t::mode::WARN:
-            return format("SimpleStrategy replication class is not "
-                "recommended, but was used for keyspace {}. The "
-                "restrict_replication_simplestrategy configuration option "
-                "can be changed to silence this warning or make it into an error.",
-                keyspace);
-        case db::tri_mode_restriction_t::mode::FALSE:
-            // Scylla was configured to allow SimpleStrategy, but let's warn
-            // if it's used on a cluster which *already* has multiple DCs:
-            if (topology.get_datacenter_endpoints().size() > 1) {
-                return "Using SimpleStrategy in a multi-datacenter environment is not recommended.";
-            }
-            break;
-        }
+
+    auto replication_strategy = locator::abstract_replication_strategy::create_replication_strategy(
+            locator::abstract_replication_strategy::to_qualified_class_name(
+                    *attrs.get_replication_strategy_class()), {})->get_type();
+    auto rs_warn_list = qp.db().get_config().replication_strategy_warn_list();
+    auto rs_fail_list = qp.db().get_config().replication_strategy_fail_list();
+
+    if (auto present_on_fail_list = std::find(rs_fail_list.begin(), rs_fail_list.end(), replication_strategy); present_on_fail_list != rs_fail_list.end()) {
+        ++stats.replication_strategy_fail_list_violations;
+        throw exceptions::configuration_exception(format(
+                "{} replication class is not recommended, and forbidden by the current configuration, "
+                "but was used for keyspace {}. You may override this restriction by modifying "
+                "replication_strategy_fail_list configuration option to not list {}.",
+                *attrs.get_replication_strategy_class(), keyspace, *attrs.get_replication_strategy_class()));
     }
+    if (auto present_on_warn_list = std::find(rs_warn_list.begin(), rs_warn_list.end(), replication_strategy); present_on_warn_list != rs_warn_list.end()) {
+        ++stats.replication_strategy_warn_list_violations;
+        return format("{} replication class is not recommended, but was used for keyspace {}. "
```

Comment:
I'm not sure this is correct. Below this `if` there sits another set of checks that could throw if the r.s. was on the warn-list but other parameters didn't meet the requirements. Presumably this did work before this check was purely for SimpleStrategy, but now when it's generic list it looks broken.

Reply:
````
error: unrecognised option '--unknown-option=true'
```

I guess we validate command line options differently than config options in scylla.yaml

Reply:
> I'm not sure this is correct. Below this `if` there sits another set of checks that could throw if the r.s. was on the warn-list but other parameters didn't meet the requirements. Presumably this did work before this check was purely for SimpleStrategy, but now when it's generic list it looks broken.

Yes, and not only that, but there also can be a ks with both RF and RS that we should warn about, in which case we'd warn only about the strategy:
```
cqlsh> CREATE KEYSPACE tracking WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor': 1};

Warnings :
SimpleStrategy replication class is not recommended, but was used for keyspace tracking. You may suppress this warning by delisting SimpleStrategy from replication_strategy_warn_list configuration option, or make it into an error by listing this replication strategy on replication_strategy_fail_list.

cqlsh> 
```

I guess both warnings have to be combined. I'll fix the code and write tests for both scenarios.
Nice catch btw. ;)

Reply:
Fixed. I tweaked the code and added 3 tests to ensure we're:
1. returning 2 warnings if both RF and replication strategy guardrails are violated
2. failing the query if RF guardrail's hard limit is violated and RS guardrail's soft limit is violated
3. vice-versa